clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -907,83 +1023,12 @@ protected void leaderExecuteTopicSwitch(
     final PubSubTopic currentLeaderTopic =
         partitionConsumptionState.getOffsetRecord().getLeaderTopic(pubSubTopicRepository);
     final PubSubTopicPartition sourceTopicPartition = partitionConsumptionState.getSourceTopicPartition(newSourceTopic);
-    Map<String, Long> upstreamOffsetsByKafkaURLs = new HashMap<>(topicSwitch.sourceKafkaServers.size());
-
     List<CharSequence> unreachableBrokerList = new ArrayList<>();
-    topicSwitch.sourceKafkaServers.forEach(sourceKafkaURL -> {
```

Comment:
This `leaderExecuteTopicSwitch` function is very long so even in the absence of needing to call this code from somewhere else, it's nice to break it up into a separate function. Thanks!